### PR TITLE
Cache the ChannelHandlerContext used in Http2StreamChannelBootstrap

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

At the moment we lookup the ChannelHandlerContext used in Http2StreamChannelBootstrap each time the open(...) method is invoked. This is not needed and we can just cache it for later usage.

Modifications:

Cache ChannelHandlerContext in volatile field.

Result:

Speed up open(...) method implementation when called multiple times

### CODE REVIEW

File: codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java [hunk 1]
```diff
@@ -137,13 +130,37 @@ public void run() {
...
         return promise;
     }
 
+    private ChannelHandlerContext findCtx() throws ClosedChannelException {
+        // First try to use cached context and if this not work lets try to lookup the context.
+        ChannelHandlerContext ctx = this.multiplexCtx;
+        if (ctx != null && !ctx.isRemoved()) {
+            return ctx;
+        }
+        ctx = channel.pipeline().context(Http2MultiplexCodec.class);
```
Comment (doom369) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java]: 2 calls to `channel.pipeline()` could be moved to a variable

Comment (normanmaurer) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java]: Sure I can do... that said I think it not really matters

Comment (normanmaurer) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java]: @doom369 addressed... please re-review.

File: codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java [hunk 2]
```diff
@@ -45,6 +45,9 @@
     private final Channel channel;
     private volatile ChannelHandler handler;
 
+    // Cache the ChannelHandlerContext to speed up open(...) operations.
+    private volatile ChannelHandlerContext multiplexCtx;
```
Comment (Scottmitch) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java]: Can you clarify if it is expected that the pipeline initialized with `Http2MultiplexCodec` is expected after the constructor (do we need/want this for 5.0 or is this just for compatibility)?

Comment (normanmaurer) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamChannelBootstrap.java]: I think it is just for compatibility for now. We can reconsider for Netty 5.

**Pull request merged**